src/main/scala/com/evolutiongaming/cluster/MappedAllocationStrategy.scala

```diff
@@ -0,0 +1,97 @@
+package com.evolutiongaming.cluster
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.cluster.sharding.ShardCoordinator.ShardAllocationStrategy
+import akka.cluster.sharding.ShardRegion
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.collection.immutable
+import scala.concurrent.Future
+
+class MappedAllocationStrategy(
+  typeName: String,
+  fallbackStrategy: ShardAllocationStrategy,
+  proxy: ActorRef,
```

Comment:
`proxy` is an `Actor` it required to be an `Actor` to be able to work with DD. So, calling methods of it directly probably is not a very good idea.